src/rideables/Stack.hpp

```diff
@@ -0,0 +1,71 @@
+// #ifndef STACK_HPP
+// #define STACK_HPP
+
+#include <stdio.h>
+#include<iostream>
+#include <atomic>
+
+using namespace std;
+
+class Stack
+{
+    private:
+
+    typedef struct StackNode
```

Comment:
There's no correctness issue, but the syntax here is too C-style. In C++, you can think `struct` as analogous to `class`, except that by default, members defined in `struct` are `public`, while in `class` they are `private`.
So the code here can be:
```c++
struct StackNode{
    ...
};
```
A pointer can then simply be `StackNode*`